MaxText/layers/linears.py

```diff
@@ -50,6 +51,13 @@
 Quant = quantizations.AqtQuantization
 
 
+def _get_model_call_mode(config):
```

Comment:
We probably could put this util [file](https://github.com/AI-Hypercomputer/maxtext/blob/6279bf1b0b211a28cb82e3d5442bdbeace023866/MaxText/max_utils.py#L4) that can be reused in other classes. Also we could have assertion to check if any other invalid value. Currently only accept `inference` I think.

Reply:
Discussed offline and added assertions in pyconfig.

Reply:
Can you reuse the the existing model_mode thats gets passed at the top level -- https://github.com/AI-Hypercomputer/maxtext/blob/main/MaxText/layers/mistral.py#L68 ?